web-console/src/druid-models/native-json-query/native-json-query-completions.ts

```diff
@@ -0,0 +1,1102 @@
+  {
+    path: '$',
+    isObject: true,
+    condition: obj => obj.queryType === 'dataSourceMetadata',
+    completions: [
+      { value: 'dataSource', documentation: 'The data source to get metadata for' },
+      { value: 'context', documentation: 'Query context parameters' },
+    ],
+  },
+
+  // Granularity simple values
+  {
+    path: '$.granularity',
+    completions: [
+      { value: 'all', documentation: 'All data in a single bucket' },
+      { value: 'none', documentation: 'No time bucketing' },
+      { value: 'second', documentation: 'Bucket by second' },
+      { value: 'minute', documentation: 'Bucket by minute' },
+      { value: 'fifteen_minute', documentation: 'Bucket by 15 minutes' },
+      { value: 'thirty_minute', documentation: 'Bucket by 30 minutes' },
+      { value: 'hour', documentation: 'Bucket by hour' },
+      { value: 'day', documentation: 'Bucket by day' },
+      { value: 'week', documentation: 'Bucket by week' },
+      { value: 'month', documentation: 'Bucket by month' },
+      { value: 'quarter', documentation: 'Bucket by quarter' },
+      { value: 'year', documentation: 'Bucket by year' },
+    ],
+  },
+
+  // Granularity object properties (when granularity is an object)
+  {
+    path: '$.granularity',
+    isObject: true,
+    condition: obj => typeof obj === 'object' && obj !== null,
+    completions: [
+      { value: 'type', documentation: 'Type of granularity' },
+      { value: 'period', documentation: 'ISO 8601 period' },
+      { value: 'timeZone', documentation: 'Timezone for bucketing' },
+      { value: 'origin', documentation: 'Origin timestamp' },
+    ],
+  },
+
+  // Granularity types
+  {
+    path: '$.granularity.type',
+    completions: [
+      { value: 'duration', documentation: 'Duration-based granularity' },
+      { value: 'period', documentation: 'Period-based granularity' },
+      { value: 'uniform', documentation: 'Uniform granularity' },
+    ],
+  },
+
+  // Duration granularity properties
+  {
+    path: '$.granularity',
+    isObject: true,
+    condition: obj => obj.type === 'duration',
+    completions: [
+      { value: 'duration', documentation: 'Duration in milliseconds' },
+      { value: 'origin', documentation: 'Origin timestamp' },
+    ],
+  },
+
+  // Period granularity properties
+  {
+    path: '$.granularity',
+    isObject: true,
+    condition: obj => obj.type === 'period',
+    completions: [
+      { value: 'period', documentation: 'ISO 8601 period string' },
+      { value: 'timeZone', documentation: 'Timezone for bucketing' },
+      { value: 'origin', documentation: 'Origin timestamp' },
+    ],
+  },
+
+  // Aggregation properties
+  {
+    path: '$.aggregations.[]',
+    isObject: true,
+    completions: [
+      { value: 'type', documentation: 'Type of aggregation' },
+      { value: 'name', documentation: 'Output name for this aggregation' },
+    ],
+  },
+
+  // Aggregation types
+  {
+    path: '$.aggregations.[].type',
```

Comment:
i've noticed this repeated quite a lot across the completions (and also some other common components like filters), is there a way to share it perhaps? fine to do as a follow-up, but sharing common elements across query, ingestion, compaction, etc would help make this a ton more maintainable